addons/website/models/ir_asset.py

```diff
@@ -58,3 +58,38 @@ def filter_duplicate(self):
                     most_specific_assets += asset
 
         return most_specific_assets
+
+    def write(self, vals):
+        '''COW for ir.asset. This way editing websites does not impact other
+        websites. Also this way newly created websites will only
+        contain the default assets.
+        '''
+        current_website_id = self.env.context.get('website_id')
+        if not current_website_id or self.env.context.get('no_cow'):
+            return super(IrAsset, self).write(vals)
+
+        for asset in self.with_context(active_test=False):
+
+            # No need of COW if the asset is already specific
+            if asset.website_id:
+                super(IrAsset, asset).write(vals)
+                continue
+
+            # If already a specific asset for this generic asset, write on it
+            website_specific_asset = asset.search([
+                ('key', '=', asset.key),
+                ('website_id', '=', current_website_id)
+            ], limit=1)
+            if website_specific_asset:
+                super(IrAsset, website_specific_asset).write(vals)
+                continue
+
+            # Set key to avoid copy() to generate an unique key as we want the
+            # specific view to have the same key
+            copy_vals = {'website_id': current_website_id, 'key': asset.key}
+
+            website_specific_asset = asset.copy(copy_vals)
```

Comment:
I had a tab open with this PR, just saw QSM asked for a change, so I did not review the PR but I saw this, that could probably already be removed for the next review.
```suggestion
            website_specific_asset = asset.copy({'website_id': current_website_id})
```
The key stuff is just to prevent an uniq behavior related to only ir.ui.view which generates an uniq key on copy.
This is most likely not needed here

Reply:
as discussed on Discord, actually we also need to keep it for 'ir.asset'
`key = fields.Char(copy=False, help='Technical field used to resolve multiple assets in a multi-website environment.')`